### ISSUE TITLE
[v4.b1] `Polyglot.transformPhrase expects argument #1 to be string` when optionText='id' in SelectInput

### ISSUE THREAD

**Issue by whg517** [marmelab/react-admin#7290]
**What you were expecting:**

When `optionText='id'` is used in the reference `SelectInput` on the form when creating an entity, the entry in the `SelectInput` should be a list of ids for the reference object

**What happened instead:**

When creating entities, an error occurs when `optionText="id"` is used in the reference `SelectInput` in the form.

```
TypeError
Polyglot.transformPhrase expects argument #1 to be string
```

**Steps to reproduce:**

**Related code:**

Error demo: [https://codesandbox.io/s/fervent-lewin-5c0wf8?file=/src/comments/CommentCreate.tsx](https://codesandbox.io/s/fervent-lewin-5c0wf8?file=/src/comments/CommentCreate.tsx)

```
<Create {...props}>
      <SimpleForm>
        <ReferenceInput
          source="post_id"
          reference="posts"
          allowEmpty
          validate={required()}
        >
          <SelectInput optionText="id" />
        </ReferenceInput>
        <DateInput source="created_at" />
        <TextInput source="body" />
      </SimpleForm>
    </Create>
```

**Other information:**

```
TypeError
Polyglot.transformPhrase expects argument #1 to be string
transformPhrase
[https://5c0wf8.csb.app/node_modules/node-polyglot/index.js:203:11]()
Polyglot.t
[https://5c0wf8.csb.app/node_modules/node-polyglot/index.js:406:14]()
Object.translate
[https://5c0wf8.csb.app/node_modules/ra-i18n-polyglot/dist/esm/index.js:42:31]()
eval
[https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:2450:103]()
eval
[https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:5907:16]()
eval
[https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:7949:220]()
eval
[https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:7963:6]()
SelectInput
[https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:7995:16]()
renderWithHooks
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:14985:18]()
updateFunctionComponent
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:17356:20]()
beginWork
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:19063:16]()
HTMLUnknownElement.callCallback
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:3945:14]()
Object.invokeGuardedCallbackDev
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:3994:16]()
invokeGuardedCallback
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:4056:31]()
beginWork$1
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:23964:7]()
performUnitOfWork
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:22776:12]()
workLoopSync
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:22707:5]()
renderRootSync
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:22670:7]()
performSyncWorkOnRoot
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:22293:18]()
eval
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:11327:26]()
unstable_runWithPriority
[https://5c0wf8.csb.app/node_modules/scheduler/cjs/scheduler.development.js:468:12]()
runWithPriority$1
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:11276:10]()
flushSyncCallbackQueueImpl
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:11322:9]()
flushSyncCallbackQueue
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:11309:3]()
NotifyManager.batchedUpdates$1 [as batchNotifyFn]
[https://5c0wf8.csb.app/node_modules/react-dom/cjs/react-dom.development.js:22387:7]()
eval
[https://5c0wf8.csb.app/node_modules/react-query/es/core/notifyManager.js:59:16]()
```

```
The above error occurred in the <SelectInput> component:


    at SelectInput (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:7927:25)

    at ChoicesContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:5648:24)

    at ResourceContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:3803:24)

    at ReferenceInput (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:7746:18)

    at div

    at eval (https://5c0wf8.csb.app/node_modules/@emotion/react/dist/emotion-element-699e6908.browser.esm.js:100:44)

    at Stack (https://5c0wf8.csb.app/node_modules/@mui/material/Stack/Stack.js:92:48)

    at DefaultComponent (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:8865:25)

    at form

    at SimpleFormView (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:8845:25)

    at FormGroupsProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:5743:24)

    at FormContent (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:5795:30)

    at exports.FormProvider (https://5c0wf8.csb.app/node_modules/react-hook-form/dist/index.cjs.js:1:21207)

    at OptionalRecordContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:4686:21)

    at Form (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:5809:24)

    at SimpleForm

    at div

    at eval (https://5c0wf8.csb.app/node_modules/@emotion/react/dist/emotion-element-699e6908.browser.esm.js:100:44)

    at Paper (https://5c0wf8.csb.app/node_modules/@mui/material/Paper/Paper.js:64:43)

    at eval (https://5c0wf8.csb.app/node_modules/@emotion/react/dist/emotion-element-699e6908.browser.esm.js:100:44)

    at Card (https://5c0wf8.csb.app/node_modules/@mui/material/Card/Card.js:43:43)

    at div

    at div

    at eval (https://5c0wf8.csb.app/node_modules/@emotion/react/dist/emotion-element-699e6908.browser.esm.js:100:44)

    at CreateView (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:8368:24)

    at RecordContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:4562:24)

    at SaveContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:4577:24)

    at CreateContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:4582:24)

    at Create (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:8421:138)

    at CommentCreate

    at Routes (https://5c0wf8.csb.app/node_modules/react-router/index.js:306:18)

    at ResourceContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:3803:24)

    at Resource (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:3807:16)

    at Routes (https://5c0wf8.csb.app/node_modules/react-router/index.js:306:18)

    at ErrorBoundary (https://5c0wf8.csb.app/node_modules/react-error-boundary/dist/react-error-boundary.esm.js:52:35)

    at div

    at main

    at div

    at div

    at eval (https://5c0wf8.csb.app/node_modules/@emotion/react/dist/emotion-element-699e6908.browser.esm.js:100:44)

    at Layout (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:4837:23)

    at div

    at Routes (https://5c0wf8.csb.app/node_modules/react-router/index.js:306:18)

    at CoreAdminRoutes (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:3895:11)

    at Routes (https://5c0wf8.csb.app/node_modules/react-router/index.js:306:18)

    at CoreAdminUI (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:3940:18)

    at AdminUI (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:10594:29)

    at InnerThemeProvider (https://5c0wf8.csb.app/node_modules/@mui/system/esm/ThemeProvider/ThemeProvider.js:16:38)

    at ThemeProvider (https://5c0wf8.csb.app/node_modules/@mui/private-theming/ThemeProvider/ThemeProvider.js:32:20)

    at ThemeProvider (https://5c0wf8.csb.app/node_modules/@mui/system/esm/ThemeProvider/ThemeProvider.js:26:20)

    at ThemeProvider (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:5143:24)

    at ResourceDefinitionContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:3768:152)

    at NotificationContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:2382:24)

    at I18nContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:2400:21)

    at Router (https://5c0wf8.csb.app/node_modules/react-router/index.js:266:18)

    at HistoryRouter (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:2100:24)

    at BasenameContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:2115:24)

    at AdminRouter (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:2118:23)

    at QueryClientProvider (https://5c0wf8.csb.app/node_modules/react-query/es/react/QueryClientProvider.js:28:21)

    at StoreContextProvider (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:2345:21)

    at CoreAdminContext (https://5c0wf8.csb.app/node_modules/ra-core/dist/esm/index.js:3779:22)

    at AdminContext (https://5c0wf8.csb.app/node_modules/ra-ui-materialui/dist/esm/index.js:10607:22)

    at Admin (https://5c0wf8.csb.app/node_modules/react-admin/dist/esm/index.js:39:22)

    at App


React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary. 

    in ErrorBoundary (created by Layout)

    in Layout (created by CoreAdminRoutes)

    in CoreAdminRoutes (created by CoreAdminUI)

    in CoreAdminUI (created by AdminUI)

    in AdminUI (created by Admin)

    in Admin (created by App)

    in App
```

**Environment**

* React-admin version: 4.0.0-beta.1.0
* Last version that did not exhibit the issue (if applicable): 3.19.10
* React version: 17.0.2
* Browser: chrome
* Stack trace (in case of a JS error):

**Comment by WiXSL** [comment]
Thanks for reporting.
The problem comes from the selected `optionText` not being of type string.

**Comment by fzaninotto** [comment]
Fixed by #7291
